05-dom-events/exercise/07-distanceConverter/distanceConverter.js

```diff
@@ -0,0 +1,37 @@
+function attachEventsListeners() {
+    const input = document.getElementById('inputDistance');
+    const inputUnits = document.getElementById('inputUnits');
+    const convertButton = document.getElementById('convert');
+    const outputUnits = document.getElementById('outputUnits');
+    const output = document.getElementById('outputDistance');
+
+    const convertInputToMeters = {
+        km: x => x * 1000,
+        m: x => x,
+        cm: x => x * 0.01,
+        mm: x => x * 0.001,
+        mi: x => x * 1609.34,
+        yrd: x => x * 0.9144,
+        ft: x => x * 0.3048,
+        in: x => x * 0.0254
+    }
+
+    const convertMetersToOutputUnits = {
```

Comment:
Brilliant solution!
I usually would prefer to construct the map like this 

```javascript 
const conversionFactors = {
    km: 1000,
    m: 1,
    cm: 0.01,
    mm: 0.001,
    mi: 1609.34,
    yrd: 0.9144,
    ft: 0.3048,
    in: 0.0254,
  };
```

and use it as a `conversionFactors[outputUnits.value]`
but this also is a very good solution!